### ISSUE TITLE
Can't set language and the list is empty

### ISSUE THREAD

**Issue by zhuzilinuser** [fasheng/arch-deepin#212]
today,I update my system,but dde not have all language.

**Comment by zhuzilinuser** [comment]
![deepinscreenshot20151230101453](https://cloud.githubusercontent.com/assets/12298476/12045454/c6d99212-aede-11e5-94d1-0970b5647a83.png)
![deepinscreenshot20151230101848](https://cloud.githubusercontent.com/assets/12298476/12045455/c6db3a68-aede-11e5-947f-ef7e8d7851a0.png)

**Comment by fasheng** [comment]
@felixonmars 
For the langauage list issue, the developer said we lost `/usr/share/i18n/{i18n_dependent.json,language_info.json}` which were added to **deepin-desktop-base** recently. But I add them manually and issue still there, so need more debugging later.

BTW: deepin-desktop-base became more and more functions now, maybe we need port it completely.

**Comment by felixonmars** [comment]
Yes, I'm already considering to make it a new package.

**Comment by felixonmars** [comment]
Update: deepin-desktop-base is a new package now.

**Comment by oberon2007** [comment]
Language list is still empty ...

**Comment by felixonmars** [comment]
Yeah, `i18n_dependent.json` defines packages to install for each locale and that's completely broken on Arch.

**Comment by zhuzilinuser** [comment]
@felixonmars  I install deepin-desktop-base,but.....list is still empty.

**Comment by adderly** [comment]
@felixonmars 
I get this in my log. I have updated every deepin package to date. If you need any debugging done, i could make that happen. As a user of course, no gdb or strace :dancer: 
*Jan 06 11:50:07 username com.deepin.daemon.LangSelector[3376]: <warning> unix64.S:122: open /usr/share/i18n/SUPPORTED: no such file or directory*

**Comment by fasheng** [comment]
Fixing now, just for the missing **/usr/share/i18n/SUPPORTED** file in archlinux.

**Comment by fasheng** [comment]
@jouyouyun Help ;)

**Comment by fasheng** [comment]
I'm not sure when it will be solved upstream. But here is a workaround temporary, just change the language manually and re-login to see the result(please correct the langauge name in the following command) 
> echo 'LANG=es_AR.UTF-8' > ~/.pam_environment

**Comment by oberon2007** [comment]
I have just tried that, with you example and with fr_FR and it does nothing! :-/

**Comment by oberon2007** [comment]
I have now manually edited **/etc/locale.gen** and then ran **locale-gen**. Now I have the language packs for applications in German and English. But Control-Center is still in English...

**Comment by fu-sen** [comment]
.pam_environment may be empty. Thereby regain the original language settings.

```
> ~/.pam_environment
```

**Comment by oberon2007** [comment]
```
$ cat ~/.pam_environment 
LANG=de_DE.UTF-8
```

**Comment by fasheng** [comment]
Well, I read the code again, found that the previous workaround not works now for dde-session-daemon will just ignore the language that not contained by the SUPPORTED language list. With the same reason, the langauge list is empty in archlinux...
Just ignore it. xD

**Comment by oberon2007** [comment]
Could we not just have **/usr/share/i18n/SUPPORTED** created somehow?

**Comment by fasheng** [comment]
From the PKGBUILD of glibc,we could see archlinux dump **SUPPORTED** to **locale.gen**
``` sh
  # create /etc/locale.gen
  install -m644 ${srcdir}/locale.gen.txt ${pkgdir}/etc/locale.gen
  sed -e '1,3d' -e 's|/| |g' -e 's|\\| |g' -e 's|^|#|g' \
    ${srcdir}/glibc-${pkgver}/localedata/SUPPORTED >> ${pkgdir}/etc/locale.gen
```

And debian did similar things, How about read supported languages from **locale.gen** as a second choice? @jouyouyun

https://projects.archlinux.org/svntogit/packages.git/tree/trunk/PKGBUILD?h=packages/glibc

**Comment by fasheng** [comment]
Solved in upstream, let's wait for **dde-api 3.0.4** now and **dde-daemon** needs to be rebuilt then.

**Comment by oberon2007** [comment]
fixed for me with today's update :)
